starknet_py/net/signer/stark_curve_signer.py

```diff
@@ -56,33 +59,35 @@ def public_key(self) -> int:
 
     def sign_transaction(
         self,
-        transaction: Transaction,
+        transaction: AccountTransaction,
```

Comment:
This is not really following the BaseSigner interface. Either interface should be updated, or we should keep the typing here as it was (I think I prefer the 2nd option).

Reply:
I think the `BaseSigner` interface should change because `Transaction` doesn't even have a `signature` field